### ISSUE TITLE
typescript syntax in js files

### ISSUE THREAD

**Issue by thecotne** [romejs/rome#514]
typescript syntax is supported in js files and it gives false impression that flow is supported and/or somehow it's correct js syntax

**Comment by sebmck** [comment]
This is a bug. What typescript syntax are you using that works? We have assertions in a bunch of places to prevent this from happening but it’s likely missing from somewhere.

**Comment by thecotne** [comment]
```ts
type Obj = {
  name: string
}

function name(obj: Obj): string {
  return obj.name
}

console.log(name({ name: 'sdsds' }))
```

<img width="440" alt="Screen Shot 2020-05-24 at 02 51 30" src="https://user-images.githubusercontent.com/1606993/82741976-8105f680-9d69-11ea-9642-142df68f90e9.png">

**Comment by sebmck** [comment]
Ahh so basically everything... the checks must have been removed by #440.

**Comment by sebmck** [comment]
Should be fixed by #521. Thank you for the report!
